### PR TITLE
switch field type to string to avoid unmarshal errors

### PR DESCRIPTION
**Pull request by marwanad**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by xizha162 — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json [hunk 1]
```diff
@@ -1997,7 +1997,7 @@
           "type": "object",
           "properties": {
             "balance-similar-node-groups": {
-              "type": "boolean"
```
Comment (majastrz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-03-01/managedClusters.json:2000]: >boolean [](start = 23, length = 7)

I discussed this over IM and while this appears to be a breaking change, this property never worked right in the service. As a result, I'm considering this more of a swagger bug fix.

From past API reviews, I recall this was basically a property bag of string properties, so coercing the bool into a string is fine here.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Review by fengzhou-msft — APPROVED** [review]

**Comment by fengzhou-msft** [comment]
@NullMDR Please check the above comment about the breaking change and force merge the PR.

**Comment by NullMDR** [comment]
@akning-ms Please help to merge this PR with breaking change approved

**Comment by aristosvo** [comment]
I don't understand why this is merged and not handled in the backend. The solution is now customer-facing and non-consistent with the other boolean options in the api specs.

**Comment by marwanad** [comment]
@aristosvo I agree and apologize for the inconvenience. In hindsight, it would've been better to handle on the backend. 

To answer your question about typing, we can't enforce Go's `time.Duration` in Swagger so the generated SDK would have to resort to `string` type. The best-case scenario would be a mix of strings with validations (for all the duration ones) and a boolean (for the balance one). Almost all the flags were durations and strings, hence the decision. We will re-evaluate this as we open up more config options.

FYI, you should be able to use v42.0.0 for the SDK which contains the fix to unblock your PR.

**Pull request merged**